cs2fs/cs2fs.fs

```diff
@@ -416,15 +416,15 @@ let rec convertNode tryImplicitConv (model: SemanticModel) (node: SyntaxNode) :
             | "--" -> res, withOp "-" "1"
             | "!" -> res, ExprApp(node, mkExprVal node "not", e)
             | "-" -> res, ExprApp(node, mkExprVal node "-", e)
-            | x -> ErrorType.UnknownPrefixOperator x |> createError node |> addError res, Mk.mkError node
+            | x -> ErrorType.UnknownPrefixOperator x |> createError node |> addError res, Mk.mkError node (sprintf "UnknownPrefixOperator %s" x)
         | PostfixUnaryExpressionSyntax(e,SyntaxToken op) as n ->
             //TODO: correct postfix operator sequence
             let res, e = descend e
             let withOp o c = [ExprInfixApp (node, e, ValId "<-", ExprInfixApp (node, e, ValId o, ExprConst (node, ConstId c))); e] |> sequence node
             match op with
             | "++" -> res, withOp "+" "1"
             | "--" -> res, withOp "-" "1"
-            | x -> ErrorType.UnknownPostfixOperator x |> createError node |> addError res, Mk.mkError node
+            | x -> ErrorType.UnknownPostfixOperator x |> createError node |> addError res, Mk.mkError node (sprintf "UnknownPostfixOperator %s" x)
```

Comment:
I actually kind of left that out on purpose, basically you should always have a corresponding error with details and you can always correlate back (because of the node) if needed?

But I have no problem with the message, just feels duplicated. I guess you plan to use that somehow in the output generation? Then I completely agree that we need something here, but It could be a 'DU' as well?

Reply:
I wanted some way to have error details in output instead of only (* ERROR *), and went by easiest route. Maybe better will be to add `AdditionalInfo : string` to `ConvertError` and then find error by node?

Reply:
From a design perspective I'm not sure what the best approach is and if the user has any benefit in `(* XZY failed/not supported *)` (considering he will get a list of errors anyway). Maybe we could even just print the C# source here.

Errors will definitely happen that's why I think it's reasonably to start with that.
Maybe @JohBa has some ideas on how we can visualize the error case (browser & cli)?

Reply:
> Maybe better will be to add AdditionalInfo : string to ConvertError and then find error by node?

Another aspect is that I'm not sure if parsing is the correct place to decide for an definite output. I'd assume that in the future we want to represent stuff differently in the browser than in the CLI (This is the reason why I added the DU for the error case instead of an error message)

Reply:
There is cases that cant be easily rewritten to F#, for example `break`, `continue`, early `return`. From CLI POV it is important that user know this is the place where `break` was in C# code.

Reply:
@jindraivanek I've planned to address at least some of them, but I agree.

Reply:
I will rework this to use ConvertResult and remove message from ExprError.